### ISSUE TITLE
[SCIP]: bliss, boost

### ISSUE THREAD

**Issue by benlorenz** [JuliaPackaging/Yggdrasil#5651]
We would like to enable the SCIP interface (and maybe soplex as well) for our `polymake_jll` but I have a few questions:

- The latest SCIP_jll contains `bin/bliss`, `include/bliss/`, and `lib/libbliss.a` which might cause conflicts when used together with `bliss_jll` ?
  This is probably because of a [change in SCIP 8.0.1](https://scipopt.org/doc-8.0.2/html/RN80.php):
  > BLISS is now built from the vendored version from https://github.com/ds4dm/Bliss
- Is there a reason not to enable boost? The [`Dependency`](https://github.com/JuliaPackaging/Yggdrasil/blob/51a1e23b487b58633deeac9ecc36b36377b0063d/S/SCIP/build_tarballs.jl#L62) is still listed anyway. We would need the boost interface to use rational arithmetic in soplex.
- <del>Under which license is SCIP distributed here?
  The [license](https://github.com/scipopt/scip/blob/master/COPYING) in the GitHub repository mandates user tracking and only applies to members academic institutions which is probably not OSI-compatible:
  ```
  1. This license applies to you only if you are a member of a noncommercial 
     and academic institution, e.g., a university. The license expires as
     soon as you are no longer a member of this institution. 
  [...]
     c. You must keep track of access to the Program (e.g., similar to the 
        registration procedure at ZIB). 
  ```
  </del>
  Resolved: SCIP is now distributed under the Apache 2.0 license

**Comment by lkastner** [comment]
Pinging @matbesancon as he seems to be the original author and might know more.

**Comment by odow** [comment]
> Under which license is SCIP distributed here? The [license](https://github.com/scipopt/scip/blob/master/COPYING) in the GitHub repository mandates user tracking and only applies to members academic institutions which is probably not OSI-compatible

That link is the correct license. SCIP is source-available. It is not open-source.

**Comment by matbesancon** [comment]
yes that's correct (for now)

**Comment by matbesancon** [comment]
> Is there a reason not to enable boost? The [Dependency](https://github.com/JuliaPackaging/Yggdrasil/blob/51a1e23b487b58633deeac9ecc36b36377b0063d/S/SCIP/build_tarballs.jl#L62) is still listed anyway. We would need the boost interface to use rational arithmetic in soplex.

When I tried building with boost it errored, maybe the issues are now fixed and we could build with it now, PR welcome to try

**Comment by odow** [comment]
Per https://github.com/scipopt/scip/blob/3c0ca57124a1a19f621d0d3847f5c8b0dc7e858e/COPYING#L89-L90

> 8. If you wish to incorporate parts of the Program into other programs whose 
>  distribution conditions are different, write to ZIB to ask for permission. 

My understanding is that ZIB is happy for us to compile and distribute `SCIP_jll` through Julia. They officially maintain the SCIP.jl Julia package: https://github.com/scipopt/SCIP.jl

They similarly maintain and distribute PySCIPOPT through conda https://github.com/scipopt/PySCIPOpt https://github.com/conda-forge/pyscipopt-feedstock

> yes that's correct (for now)

The "for now" also hints that ZIB are considering a potential relicensing under an OSI license. But they haven't done so to date.

> anyone installing SCIP_jll who is not a member of an academic institution will also violate that license (probably without even noticing)

The only packages that depend on SCIP_jll are SCIP.jl and Boscia.jl, both developed by ZIP: https://juliahub.com/ui/Packages/SCIP_jll/Qn2LI/800.0.200+1?page=2

The only package that depends on SCIP is Boscia: https://juliahub.com/ui/Packages/SCIP/a0EUw/0.11.8?page=2

If you write a package that depends on SCIP, you should do due diligence on the license restrictions. It also doesn't install automatically on Windows, which likely limits the impact. 

If you are a _user_ of SCIP in a commercial setting, the same license due diligence restrictions apply. But at minimum, we should clarify this in the README: https://github.com/scipopt/SCIP.jl/pull/250

> This afaict contradicts the policy https://github.com/JuliaRegistries/General/issues/36269.

Yes, this is one of those  "I'll also note that there are probably a few such instances already"

**Comment by matbesancon** [comment]
> The "for now" also hints that ZIB are considering a potential relicensing under an OSI license. But they haven't done so to date.

This is being done at the moment (there were minor issues that are being checked), it is a matter of weeks more, the agreement on principle is there.

**Comment by matbesancon** [comment]
@benlorenz this is technically not merged into the version in Yggdrasil yet but SCIP is now open-source under Apache 2.0, I think this issue can be closed :)

**Comment by matbesancon** [comment]
Ah indeed I had forgotten about the other part. For bliss we vendored it by default to fix to 0.77 and for simplicity for users compiling themselves.  We could in the build script not take the included one but bliss_jll, for boost it could be included yes I don't think there is a blocker
